### PR TITLE
PVR/EPG code cleanup

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

**Comment by MaxKellermann** [comment]
See https://github.com/xbmc/xbmc/pull/9414 for a discussion about this Travis CI failure.

**Comment by MaxKellermann** [comment]
Unit test crashes due to https://github.com/xbmc/xbmc/pull/9389 and not due to this PR.

**Comment by Jalle19** [comment]
Looks good to me. Ping @ksooo

**Comment by MaxKellermann** [comment]
Rebased after commit f711253

**Comment by MaxKellermann** [comment]
There was a heap corruption in one Travis-CI build, but it doesn't look related to this PR.

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by ksooo** [comment]
+1. thx @MaxKellermann

**Comment by MaxKellermann** [comment]
Rebased & fixed the missing include for size_t.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/epg/GUIEPGGridContainer.cpp [hunk 1]
```diff
@@ -1644,7 +1644,7 @@ GridItemsPtr *CGUIEPGGridContainer::GetPrevItem(const int &channel)
   return &m_gridIndex[channelIndex][i + m_blockOffset];
 }
 
-GridItemsPtr *CGUIEPGGridContainer::GetItem(const int &channel)
+GridItemsPtr *CGUIEPGGridContainer::GetItem(const int channel)
```
Comment (ksooo) [inline xbmc/epg/GUIEPGGridContainer.cpp]: const (here any dabove)? nope. does not even match declaraion anmore. ;-)

Comment (MaxKellermann) [inline xbmc/epg/GUIEPGGridContainer.cpp]: The "const" here does not need to match the prototype - it's not part of the API. It is a local declaration for the code inside the method, asking the compiler to forbid all modifications. It has no effect outside the method body.
Do you want me to remove it anyway?

Comment (ksooo) [inline xbmc/epg/GUIEPGGridContainer.cpp]: > The "const" here does not need to match the prototype - it's not part of the API. It is a local declaration for the code inside the method, asking the compiler to forbid all modifications. It has no effect outside the method body.

 thanks for explaining. did not know this.

> Do you want me to remove it anyway?

 yes, please do - for the sake of consistency. I guess there are more people like me out there, stumbling over this kind of "mismatch".

**Comment by MaxKellermann** [comment]
Removed a few unnecessary "const" keywords as requested by @ksooo

**Comment by ksooo** [comment]
thx @MaxKellermann 

jenkins build this please

**Comment by MaxKellermann** [comment]
Jenkins failed on OS X because it failed to download libtool. Sigh. Something's very wrong with automatic builds currently.

File: xbmc/epg/GUIEPGGridContainer.cpp (hunk 1 above)
Comment (Jalle19) [inline xbmc/epg/GUIEPGGridContainer.cpp]: I didn't even know that was possible.

**Comment by FernetMenta** [comment]
ignore the build errors. I'll try to find an admin to upload libtool

**Comment by ksooo** [comment]
@MaxKellermann WIN-32 jenkins build fails for a real reason, I'd say. sigh.

c:\jenkins\slave\workspace\win-32\xbmc\epg\GUIEPGGridContainer.h(43): error C2146: syntax error : missing ';' before identifier 'size_t' (..\..\xbmc\epg\GUIEPGGridContainer.cpp) [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]
         c:\jenkins\slave\workspace\win-32\xbmc\epg\GUIEPGGridContainer.h(43): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (..\..\xbmc\epg\GUIEPGGridContainer.cpp)

**Comment by MaxKellermann** [comment]
@ksooo I fixed that problem already (see https://github.com/xbmc/xbmc/pull/9415#issuecomment-199724569); this is an old build.

**Comment by MaxKellermann** [comment]
Oh, I was wrong here. The WIN32 problem was not a missing include for size_t, but the inability of the WIN32 compiler to understand the standard C++11 keyword `constexpr`. D'oh!

**Comment by MaxKellermann** [comment]
Converted `constexpr` to `const`. I hope this works on WIN32 now.

Anyway, why are you using an outdated compiler on Windows? VS2010 is obviously older than C++11..

**Comment by Montellese** [comment]
We are using VS2013 which doesn't support constexpr and a few other things. But AFAIK not even the latest VC++ compiler shipped with VS2015 has full C++11/14 support.

**Comment by ksooo** [comment]
well...
jenkins build this please

**Comment by MaxKellermann** [comment]
Indeed, Jenkins builds Kodi with VS2013 (12.0); I have been misled by the many appearances of "VS2010" in the Jenkins log, but that's just a directory name inside the Kodi repository.
VS2015 (14.0) does support `constexpr`: https://msdn.microsoft.com/en-us/library/dn956974.aspx
`constexpr` is a great feature.

**Comment by MaxKellermann** [comment]
Same WIN32 build failure with `const`:

    c:\jenkins\slave\workspace\win-32\xbmc\epg\GUIEPGGridContainer.h(43): error C2146: syntax error : missing ';' before identifier 'size_t' (..\..\xbmc\epg\GUIEPGGridContainer.cpp) [c:\jenkins\slave\workspace\WIN-32\project\VS2010Express\XBMC.vcxproj]

What does that mean? The code looks like valid C++03, and `stddef.h` was included to make `size_t` known to the compiler.

**Comment by afedchin** [comment]
odd, I've builded with this condition locally w/out any errors.

**Comment by ksooo** [comment]
May I suggest to drop that constexpr commit for now and to solve this in another PR? The rest could be merged immediately.

**Comment by MaxKellermann** [comment]
OK, dropped, keeping it in my master branch.

**Comment by ksooo** [comment]
Thx, let's give it (the final) try:

jenkins build this please

**Comment by MaxKellermann** [comment]
Yay, WIN32 "Finished: SUCCESS".

**Pull request merged**